discovery_engine_core/core/src/engine.rs

```diff
@@ -517,54 +518,95 @@ impl Engine {
         &mut self,
         history: Option<&[HistoricDocument]>,
         sources: &[WeightedSource],
-        reacted: &UserReacted,
-    ) -> Result<(), Error> {
+        reacted: UserReacted,
+    ) -> Result<Document, Error> {
+        let reaction = reacted.reaction;
+        cfg_if! {
+            if #[cfg(feature="storage")] {
+                let document: Document = self.storage
```

Comment:
The type annotation is not strictly needed, but makes things a bit nicer when juggling features.